src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+  icon,
+  size = "default",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  if (forceActiveState && forceHoverState) {
+    throw new TypeError(
+      "Do not force multiple properties at once, you will get indeterminiate behavior"
+    );
+  }
+
+  // We're storing this because it can be used for the `:active` style or when
+  // `forceActiveState` is used in Storybook
+  const activeProperties = {
+    ...(hidden
+      ? { backgroundColor: hiddenActiveBackgroundColor }
+      : {
+          // The `box-shadow` property is copied directly from Zeplin
+          boxShadow:
+            "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)"
+        }),
+    outline: 0
+  };
+
+  // We're storing this because it can be used for the `:hover` style or when
+  // `forceHoverState` is used in Storybook
+  const hoverProperties = {
+    backgroundColor: hoverBackgroundColor,
+    cursor: "pointer",
+    ...(!hidden && {
+      // The `box-shadow` property is copied directly from Zeplin
+      boxShadow:
+        "0 5px 10px 0 rgba(18, 21, 26, 0.12), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+    })
+  };
+
+  // We're storing this because it can be used for the `:focus` style or when
+  // `forceFocusState` is used in Storybook
+  const focusProperties = {
+    // The `box-shadow` property is copied directly from Zeplin
+    boxShadow:
+      "0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 0 0 2px #bbdbff, inset 0 0 0 1px #2075d6, inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)",
+    ...(hidden && { color: `${hiddenFocusedColor} !important` })
+  };
+
+  return (
+    <button
+      {...otherProps}
+      css={{
+        appearance: "button",
+        backgroundColor: hidden
+          ? "transparent"
+          : disabled
+          ? disabledBackgroundColor
+          : backgroundColor,
+        borderRadius: fab ? "100%" : 4,
+        borderWidth: 0,
+        ...(!hidden && {
+          boxShadow:
+            "0 1px 4px 0 rgba(18, 21, 26, 0.08), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+        }),
+        minWidth: iconOnly
+          ? size === "small"
+            ? 28
+            : size === "large"
+            ? 42
+            : 36
+          : size === "small"
+          ? 76
+          : size === "large"
+          ? 112
+          : 100,
+        padding: size === "small" ? `5px 8px` : size === "large" ? 8 : 7,
+        ...(size === "small"
```

Comment:
Similar to the comment above about `minWidth`, we could do something like this here:

```js
function getFontStyles(size) {
  switch(size) {
    case 'small':
      return base.small;
    case 'large':
      return base.large;
    default:
      return base.base;
  }
}

{
  ...getFontStyles(size)
}
```

Alternatively, since the size prop lines up with the keys of the typography `base` we could do something like this:

```js
{
  ...base[size || 'base']
}
```